TSE required changes

sfdrogojan opened the pull request.

sfdrogojan committed:
Using fueloauth SOAP authentication

sfdrogojan committed:
Merge branch 'dev' of https://github.com/sfdrogojan/FuelSDK-CSharp into fueloauth-authentication-refactoring

sfdrogojan committed:
Removed default value for SOAP endpoint and fetched it only if not present in the config file.

sfdrogojan committed:
Merge branch 'dev' of https://github.com/sfdrogojan/FuelSDK-CSharp into fueloauth-authentication-refactoring

sfdrogojan committed:
Fixed the code for missing legacy token.

Commit:
Merge branch 'soap-endpoint-fetch-fix' of https://github.com/sfdrogojan/FuelSDK-CSharp into TSE-required-changes

Commit:
Stack key

CCBet committed:
Added a validation

Commit:
Added the caching mechanism for the SOAP Endpoint

sfdrogojan committed:
Refactoring

sfdrogojan committed:
Fixes.

manivinesh reviewed (changes requested):
Change the default soap to s4 endpoint.
Validate the JWT flow

Commit:
Changed the default value of the SOAP endpoint to https://webservice.s4.exacttarget.com/Service.asmx

sfdrogojan committed:
Removed InternalAuthToken set when authentication uses clientId and clientSecret.

sfdrogojan committed:
Merge branch 'TSE-required-changes' of https://github.com/sfdrogojan/FuelSDK-CSharp into TSE-required-changes

manivinesh reviewed (approved):
LGTM

Commit:
Implemented the stackKey flag in QS requirement

Commit:
Fixed a SOAP endpoint cache bug

In the case when the SOAP endpoint was not specified in the App.config file, a fetch is made for this endpoint but the same endpoint was used even after the expiration date of the endpoint - fixed

Commit:
Rollback to the state of using legacy token

sfdrogojan committed:
Reverted some comments to the previous version.

sfdrogojan committed:
StackKey refactoring

sfdrogojan committed:
Legacy query builder

sfdrogojan committed:
Removed the stackKeyPropertyAccess parameter from UserInfo's constructor
